src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/SimpleConfigModel.cs

```diff
@@ -341,6 +344,19 @@ IGlobalRunConfig IRunnableProjectConfig.OperationConfig
             }
         }
 
+        private IReadOnlyList<IReplacementTokens> _symbolFilenameReplacements;
+        internal IReadOnlyList<IReplacementTokens> SymbolFilenameReplacements
+        {
+            get
+            {
+                if (_symbolFilenameReplacements == null)
+                {
+                    _symbolFilenameReplacements = ProduceSymbolFilenameReplacements();
```

Comment:
minor, is this "if null" initialization done here, because _symbolFilenameReplacements can become null beyond class instantiation?  If not, please consider the default initialization in the .ctor

Reply:
SimpleConfigModel .ctor should not be used when creating template, class can be properly instantiated via FromJObject method. In fact it should not be even public, probably somebody wanted to use it for unit tests in invalid way. 
The idea behind this approach is to prevent calculating properties when they are not used. 
For example when running dotnet new all templates in cache will be read using SimpleConfigModel, however only one of them will be processed and FilenameReplacements will be calculated only for it. 
Agree that it is probably not the perfect way to use this class, and it needs refactoring.